src/models/user.test.ts

```diff
@@ -0,0 +1,54 @@
+import PullRequest from './pull-request';
+import User from './user';
+
+const pullRequest1 = new PullRequest('1', 'testURL/1', [
+    { user: { login: 'test1' }, state: 'CHANGES_REQUESTED' },
+    { user: { login: 'test2' }, state: 'APPROVED' },
+]);
+
+const pullRequest2 = new PullRequest('2', 'testURL/1', [
+    { user: { login: 'test2' }, state: 'CHANGES_REQUESTED' },
+    { user: { login: 'test1' }, state: 'APPROVED' },
+]);
+
+const pullRequest3 = new PullRequest('3', 'testURL/1', [
+    { user: { login: 'test2' }, state: 'CHANGES_REQUESTED' },
+    { user: { login: 'test1' }, state: 'APPROVED' },
+]);
+
+describe('User', () => {
+    it('creates user with no pull requests', async () => {
+        const user = new User('test1');
+        expect(user.pullRequestsReviewed.length).toEqual(0);
+        expect(user.pullRequestsSubmitted.length).toEqual(0);
+    });
+
+    it('#getRejectsCount returns count of rejected reviews by user', async () => {
+        const user = new User('test1');
+        user.pullRequestsReviewed.push(pullRequest1);
+        user.pullRequestsReviewed.push(pullRequest2);
+        expect(user.getRejectsCount()).toEqual(1);
+    });
+
+    it('#getApprovesCount returns count of approved reviews by user', async () => {
+        const user = new User('test1');
+        user.pullRequestsReviewed.push(pullRequest1);
+        user.pullRequestsReviewed.push(pullRequest2);
+        user.pullRequestsReviewed.push(pullRequest3);
+        expect(user.getApprovesCount()).toEqual(2);
+    });
+
+    it('#getPRRatio returns ratio of submitted PRs to reviewed PRs', async () => {
+        const user = new User('test1');
+        user.pullRequestsReviewed.push(pullRequest1);
+        user.pullRequestsReviewed.push(pullRequest2);
+        user.pullRequestsSubmitted.push(pullRequest3);
+        expect(user.getPRRatio()).toEqual(2);
+    });
+
+    it('#getPRRatio works when there are no submitted PRs', async () => {
+        const user = new User('test1');
+        user.pullRequestsReviewed.push(pullRequest1);
+        expect(user.getPRRatio()).toEqual(1);
+    });
+});
```

Comment:
I used snapshot here - https://github.com/jniechcial/code-review-watchman/pull/3/files#diff-e9cc705f0363f54750b69405c74ffc96R22. The rest is a number based expect, so I left as it was.